### DISCUSSION TITLE
Architecture  - Forwarder + Elastic Agents

### DISCUSSION THREAD

**Discussion by Mav1814** [Security-Onion-Solutions/securityonion#10790] [2.4] [answered]
Hello, 

First of all, I would like to congratulate the excellent work that has been developed over the years. 

I have three questions regarding Elastic Agents. I hope I can explain myself:

1 - In the architecture, the Forwarders Nodes already have an elastic agent installed, and the components "Strelka, Suricata and Zeek" through the elastic agent forward the logs to the manager.  Is this correct?
2 - In the case of installing multiple Elastic Agents on endpoints, is this "management" of forwarding logs done by the manager or is it done by the forwarders? For example, does a W10 endpoint, after installing the Elastic Agent, forward the logs etc, directly to the manager or is it done via a forwarder? 
3 -  In case of elastic agents installed on endpoints forward the logs through the manager, is there a possibility of overloading the manager? Or the logs of these elastic agents are pulled from the Redis queued on the manager by the search nodes?

In my case, I have a manager, 5 forwarders nodes (in different networks), and two search nodes. The doubts arose due to the fact that I have forwarders in different networks (two interfaces) and I have endpoints in these networks when installing the elastic agent, the logs are forwarded to the manager or is it through the forwarders in the respective networks?

**Comment by defensivedepth** [comment, marked as answer]
For 2.4 - All Elastic Agents (whether installed on the grid nodes or endpoints) connect to the following for management & sending their logs:

- Manager
- Fleet Node
- Receiver Node (logs only, no management)

If you have more than one of these setup in your grid, Elastic Agent will use round robin load balancing. 

It is not completely documented yet, but the overall architecture would be that if you are planning on deploying a moderate to large amount of Elastic Agents, you should stand up a Fleet Node.

**Reply by Mav1814** [reply]
I appreciate the reply. I closed the questions because I eventually found Fleet Node.
